routing_model/src/flows_dst.rs

```diff
@@ -281,11 +284,41 @@ impl CheckAndProcessElderChange {
 
     fn try_consensus(&self, vote: &ParsecVote) -> Option<Self> {
         match vote {
-            ParsecVote::CheckElder => Some(self.check_elder()),
+            ParsecVote::NeighbourMerge(merge_info) => Some(self.store_merge_infos(*merge_info)),
+            ParsecVote::CheckElder => Some(self.check_merge()),
+            _ => None,
+        }
+    }
+
+    fn try_rpc(&self, rpc: Rpc) -> Option<Self> {
+        match rpc {
+            Rpc::Merge => Some(self.vote_parsec_neighbour_merge()),
             _ => None,
         }
     }
 
+    fn store_merge_infos(&self, merge_info: MergeInfo) -> Self {
+        self.0.action.store_merge_infos(merge_info);
+        self.clone()
+    }
+
+    fn merge_needed(&self) -> bool {
+        // TODO: trigger for merge needed still in flux
```

Comment:
For this PR, we would want to have this return true and false.

You can choose a criteria that is easy to implement and somewhat sensible, it does not have to relate to what the real merge needed require: I would say for now, just mock it (i.e have a field in InnerAction: merge_needed:bool